### PR TITLE
Github Actions for CRAN Release Testing and Website Doc Update

### PR DESCRIPTION
**Pull request by bburns632**
I created two github actions workflows replacing previous processes: 

#### Pre-Release DEVEL testing on Linux, Mac & Windows
This is a `CRAN` requirement for a new package release.  This streamlines the local docker setup I had before. 

#### Website Documentation Updates
When a new version tag is created in this repo, this should trigger a branch to be created with website documentation updates via `pkgdown`.  🤞 this makes docs update a one click process.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -295,71 +295,43 @@ Once substantial time has passed or significant changes have been made to `pkgne
...
-4. Once your satisfied with your results, you're done with _linux_ checks.  
-5. To test on windows r-devel (as CRAN will do), take the final package tarball and upload it to this webpage: https://win-builder.r-project.org/upload.aspx
-6. Grab a coffee and wait for an email (about 10 minutes later but varies)
-7. Fix your errors and reupload until good. 
+Once checks all pass, _Don't merge the PR quite yet!_  Follow the steps below to [submit to CRAN](https://cran.r-project.org/submit.html) which is still a minor manual process.
 
 ### Submit to CRAN
 
-Go to https://cran.r-project.org/submit.html, take the tarball that passed the gauntlet and submit it! 
+1. Build the tarball locally on the current release branch: `R CMD Build .`
```
Comment (jayqi) [inline CONTRIBUTING.md:322]: Wondering if it makes sense to do this in the CI. Then you could just download the tarball if the tests pass. 

https://github.com/actions/upload-artifact

**Review by jayqi — APPROVED** [review]
Looks good. Had an idea about running `R CMD Build` as part of the release CI workflow, but it doesn't need to block this PR.

File: CONTRIBUTING.md (hunk 1 above)
Comment (bburns632) [reply CONTRIBUTING.md:322]: That's a great call out.  TIL.  Assuming the uploaded artifact is just then available and downloadable from the run GUI/CLI?

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -45,19 +46,29 @@ jobs:
...
         run: $GITHUB_WORKSPACE/test.sh
   all-successful:
     runs-on: ubuntu-latest
-    needs: [devel_test]
+    needs: [version_check, devel_test]
     steps:
       - name: Note that all tests succeeded
         run: echo "🎉"
+      - name: build tarball for submission to CRAN
+        run: R CMD Build .
```
Comment (jameslamb) [inline .github/workflows/release.yml:63]:
```suggestion
        run: R CMD build .
```

I'm pretty sure this is lowercase.

Comment (jameslamb) [reply .github/workflows/release.yml:63]: for example: https://support.posit.co/hc/en-us/articles/200486518-Customizing-Package-Build-Options-in-the-RStudio-IDE

**Review by jameslamb — APPROVED** [review]

File: CONTRIBUTING.md (hunk 1 above)
Comment (jayqi) [reply CONTRIBUTING.md:322]: Yeah, so here's an example CI run where I'm testing a tool that generates some images, and I wanted to persist those images so I could manually look at them if I wanted to.

You can see the download links if you scroll down to the bottom to "Artifacts": https://github.com/drivendataorg/erdantic/actions/runs/8677103132

Here's the `actions/upload-artifact` step: https://github.com/drivendataorg/erdantic/blob/c687b03d3647b619788e7396043ab927afbe057d/.github/workflows/tests.yml#L65-L70

**Pull request merged**